File: Sorting/SelectionSort.java
/**
 *  Selection Sort.
 *  @author: himself
 */
public class SelectionSort {

    /**
     * Selecton Sort.
     */
    static int[] selectionSort(int[] arr) {
        int n = arr.length;
        int min, mini, tmp;

        for(int i = 0; i < n; i++) {
            min = arr[i];
            mini = i;
            for(int j = i + 1; j < n; j++) {
                if(arr[j] < min) {
                    min = arr[j];
                    mini = j;
                }
            }
            tmp = arr[i];
            arr[i] = arr[mini];
            arr[mini] = tmp;
        }

        return arr;
    }

    public static void main(String[] args) {
        // int[] arr = {-100, 100, 213, 0, 2, 932, -132, -1, 4, 0};
        // int[] sorted = selectionSort(arr);
        // for(int i = 0; i < sorted.length; i++)
        //     System.out.print(sorted[i] + " ");
    }
}

File: Sorting/C++/selection_sort.cpp
#include <bits/stdc++.h>

using namespace std;

#define MAX_N 10000

void selection_sort(int arr[], int n) {
	for (int i = 0; i < n; ++i) {
		int min_i = i;
		for (int j = i + 1; j < n; ++j) {
			if (arr[j] < arr[min_i]) {
				min_i = j;
			}
		}
		swap(arr[min_i], arr[i]);
	}
}

int main() {
   	int arr[MAX_N];
   	int n; cin >> n;
   	for (int i = 0; i < n; ++i) {
   		cin >> arr[i];
   	}

   	selection_sort(arr, n);

   	for (int i = 0; i < n; ++i) {
   		cout << arr[i] << " ";
   	}
}
